### PR TITLE
Add a flag to determine the usage of Chart and friends

### LINKED ISSUE #22: Add a flag to compile without charting support
We often do not need the charting support and it has heavy dependencies and does not move quickly with new GHC versions. For a lighter version we can make it modular such that it can be compiled without charts.

### PR DESCRIPTION
**Pull request by adithyaov**

### CODE REVIEW

File: bench-show.cabal [hunk 1]
```diff
@@ -96,13 +96,17 @@ source-repository head
     type: git
     location: https://github.com/composewell/bench-show
 
+flag charting
+  description: Build the modules that provide charting functionality
+  manual: True
+  default: False
```
Comment (harendra-kumar) [inline bench-show.cabal:102]: This should be true by default to keep it compatible with previous versions. We can name the flag no-charts to indicate that charts are on by default.

File: lib/BenchShow.hs [hunk 2]
```diff
@@ -97,5 +99,15 @@ module BenchShow
...
+#ifdef CHARTING
+import BenchShow.Graph
+#else
+graph :: FilePath -> FilePath -> Config -> IO ()
+graph _ _ _ =
+    putStrLn
+        $ unlines [ "Library was compiled without charting support."
+                  , "See the flags available in the cabal file for more info."
+                  ]
+#endif
```
Comment (harendra-kumar) [inline lib/BenchShow.hs]: We can keep this module unchanged. If we want to use it without graphs we can import the lower level modules.

Comment (adithyaov) [reply lib/BenchShow.hs]: What do you mean by unchanged? This needs to be changed if graphs do not exist right? Are you suggesting we move this CPP stuff into `Grpah` module?

Comment (harendra-kumar) [reply lib/BenchShow.hs]: This can always include the Graph module. When we have `no-charts` flag on we can exclude this module altogether and just use the lower level module directly.

File: app/Main.hs [hunk 3]
```diff
@@ -22,7 +22,13 @@ import Options.Applicative.Simple (addCommand, simpleOptions, simpleVersion)
 import Paths_bench_show (version)
 
 import Options.Applicative
+
+#ifndef NO_CHARTS
 import BenchShow
+#else
+import BenchShow.Internal.Common
+import BenchShow.Internal.Report
+#endif
```
Comment (harendra-kumar) [inline app/Main.hs:31]: It may be cleaner to put the charts CLI functionality in a separate module, and just use addCommand in this module if charts is configured. But we can do that in a separate PR.

**Review by harendra-kumar — APPROVED** [review]
